brave/brave-core
# Load Brave Ads resources basing on geo position

Resolves https://github.com/brave/brave-browser/issues/31765

Now `Brave Ads Resources` and `NTP Sponsored Images` components are selected/downloaded based on geo position which we get from https://geo.ads.brave.com/v1/getstate. The default value of geo position is an OS region.

If the response is `{"country":"US","region":"FL"}` then the following components will be downloaded:
* Brave Ads Resources (US)
* NTP Sponsored Images (US)

The functionality is under the `FetchResourcesByCountryCode` feature which is enabled by default.

The geo position is periodically fetched. The time period is controlled by the `fetch_interval_minutes` parameter of the `Subdivision` feature. The command line argument allows to set the period to 1 minute: 
`--enable-features=Subdivision:fetch_interval_minutes/1m`

The default value of the time period is 1 day.



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

* Start with a fresh profile
* Open brave://components
* Make sure that resources are loaded based on the system region settings:
  - NTP Sponsored Images
  - Brave Ads Resources
* Join Brave Rewards
* Open brave://components
* Make sure that resources are loaded based on the current https://geo.ads.brave.com/v1/getstate response:
  - NTP Sponsored Images
  - Brave Ads Resources
* If the current country is USA or Canada:
  - Brave Rewards page
  - Click the Settings button
  - Make sure that `Regional ad relevance` setting is correctly set
  - Make sure that `Regional ad relevance` can be changed
* Reset brave rewards
* Make sure that resources are loaded based on the system region settings:
  - NTP Sponsored Images
  - Brave Ads Resources
